.github/workflows/ci.yml

```
@@ -137,8 +150,21 @@ jobs:
 
     - run: nox --python ${{ matrix.python.action }} -e ${{ matrix.task.nox }} -- --use-wheel dist/*.whl
 
+    - name: Store coverage file
+      uses: actions/upload-artifact@v3
+      with:
+        # We have a single artifact shared for each workflow run.
+        # in this way we should intefere with coverage reports from other PRs
+        # or previous runs for the same PR.
```

Slight wording tweak, though I'm not sure there's really a need to have the run ID in the name.  Yes, the files need to be unique but AFAIK uploading like this uses names within a namespace for this run.  And actually, if you get a new run ID for reruns then I think this setup wouldn't work in case a job flakes and you rerun it as you would then get two separate artifacts.

```suggestion
        # We have a single artifact shared for each workflow run.
        # in this way we should not interfere with coverage reports from other PRs
        # or previous runs for the same PR.
```